flask_jwt_extended/tokens.py

```diff
@@ -138,8 +167,15 @@ def decode_jwt(encoded_token, secret, algorithm, identity_claim_key,
     if data['type'] == 'access':
         if 'fresh' not in data:
             raise JWTDecodeError("Missing claim: fresh")
+   
     if user_claims_key not in data:
         data[user_claims_key] = {}
+    
+    if data['type'] != 'refresh' or config.user_claims_in_refresh_token:
```

Comment:
Related to above, I think we shouldn't have this code path, and instead rely on the existing code path for verifying custom data in user_claims, and pyjwt.decode errors for standard JWT claims.